packages/controller/main.js

```diff
@@ -695,6 +696,24 @@ function createObjects(onConnect) {
                             delete hostAdapter[id];
                         }
                     }
+                } else if (obj.type && obj.type === 'host' && obj.common && obj.common.installedVersion) {
```

Comment:
Should we add another else (or change to detect a host object by id regex?) to detect host object deletions?

Reply:
Basically we also need to check this, so if we are running and a host is removed, also all controllers would need a restart if there is the possibility that now the newer protocol can be used.